### PR TITLE
Created a function to download and upload a file using PowerShell CLI

### PR DESCRIPTION
**Pull request by sharma-shruti**

### CODE REVIEW

File: articles/File_download.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                [Parameter(Mandatory=$true)]
+                [String]$fileName
+
+        )
+        $fileContent = Get-Content $fileName
+        $fileString = $fileContent.Split("</p>")[1] + "</p>"
+        $xmlHeader= [xml]$fileString
+        $hrefString = $xmlHeader.p.a.href
+        $key = ([System.Uri]$hrefString).Query.Split("=")[1]
+        $url = "https://fast.wistia.net/embed/iframe/" + $key
```
Comment (abhi18av) [inline articles/File_download.md]: Better name would be `$videoUrl`

File: articles/File_download.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        param(
+                [Parameter(Mandatory=$true)]
+                [String]$fileName
+
+        )
+        $fileContent = Get-Content $fileName
+        $fileString = $fileContent.Split("</p>")[1] + "</p>"
+        $xmlHeader= [xml]$fileString
+        $hrefString = $xmlHeader.p.a.href
+        $key = ([System.Uri]$hrefString).Query.Split("=")[1]
```
Comment (abhi18av) [inline articles/File_download.md]: Instead of `key` let's use `videoId`

File: articles/File_download.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        $url = "https://fast.wistia.net/embed/iframe/" + $key
+
+        # Download the video from the above URL
+        youtube-dl $url
+
+        # Get the name of the downloaded file
+        $outputFile = Get-ChildItem *.bin -Name
+        $fileString = $outputFile.Split("-")[0]+"-$key" + ".mp4"
```
Comment (abhi18av) [inline articles/File_download.md]: We could probably extract this as a separate function and simply call that function here.

File: articles/File_download.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
+# Download file from a website and upload it to Dropbox 
+
+## My very first function in powershell ~
+
+```powershell
+function Get-UrlInfo {
```
Comment (abhi18av) [inline articles/File_download.md:6]: Let's switch to a better name like `Save-Video-Wistia` perhaps?

File: articles/File_download.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        [Cmdletbinding()]
+        param(
+                [Parameter(Mandatory=$true)]
+                [String]$fileName
```
Comment (abhi18av) [inline articles/File_download.md:10]: Is there is a real need to actually rely on a `fileName` - maybe we could just use the url string as an input. What do you think?

File: articles/File_download.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+        # Get the name of the downloaded file
+        $outputFile = Get-ChildItem *.bin -Name
+        $fileString = $outputFile.Split("-")[0]+"-$key" + ".mp4"
+
+        # Rename the downloaded file
+        $renameFile = mv $outputFile $fileString
+
+        # Upload the file on to Dropbox
+        rclone copy *.mp4 remote:/Videos/ -v
```
Comment (abhi18av) [inline articles/File_download.md:31]: Better to be more specific about the actual file you want to upload rather than a general `*.mp4`.

**Review by abhi18av — CHANGES_REQUESTED** [review]
We can still make more changes for a better `UX` but this looks good for a start.

File: articles/File_download.md (hunk 5 above)
Comment (sharma-shruti) [reply articles/File_download.md:10]: No specific reason, I was just reading the content from a file and then trying to parse it. 
Using URL it should look something like this, right? (btw, I will add the url as input to my function later)
```powershell
PS /root> $inputString = "https://university.business-science.io/courses/541207/lectures/15050206?wvideo=9gj4s6uckw"
PS /root> $videoId = [System.Uri]$inputString
PS /root> $videoUrl = $videoId.Query.Split("=")[1]
PS /root> $link = "https://fast.wistia.net/embed/iframe/" + $videoUrl
PS /root> $link
https://fast.wistia.net/embed/iframe/9gj4s6uckw
PS /root> youtube-dl $link
```

File: articles/File_download.md (hunk 6 above)
Comment (sharma-shruti) [reply articles/File_download.md:31]: You must have noticed that after renaming the file from `.bin` to `.mp4`, I am storing it in a variable **$renameFile**. However, I tried using this variable to copy the file, but it didn't work.
`rclone copy $renameFile remote:/Videos/ -v`
Error - Unable to get the file path

`rclone copy ./$renameFile remote:/Videos/ -v`
Worked - but uploaded all the files and folders from the current directory

Hence, I used this option. How would you do it?

File: articles/File_download.md (hunk 4 above)
Comment (sharma-shruti) [reply articles/File_download.md:6]: Do you mean renaming the file?

File: articles/File_download.md (hunk 3 above)
Comment (sharma-shruti) [reply articles/File_download.md]: Ok. I have to try this.

File: articles/File_download.md (hunk 6 above)
Comment (abhi18av) [reply articles/File_download.md:31]: I'm not quite sure at the moment, would look into it later. As a reminder, I'm converting the above message into an `issue` as a reminder.

File: articles/File_download.md (hunk 4 above)
Comment (abhi18av) [reply articles/File_download.md:6]: Nope, renaming the function.

File: articles/File_download.md (hunk 5 above)
Comment (abhi18av) [reply articles/File_download.md:10]: Yes, that seems right. 👍 

BTW, the `input to a function` is called a `parameter` :)

File: articles/File_download.md (hunk 3 above)
Comment (sharma-shruti) [reply articles/File_download.md]: How do I extract a separate function for this, as I have to rely on `$key` variable, which means I will have to execute all the above steps first? I wonder how creating a separate function will make any difference here. What do you think?

**Comment by abhi18av** [comment]
@sharma-shruti , after making updates to the changes either `resolve` the above specific conversation or comment their that you've completed the changes or have other thoughts.

Regarding the `[fix]` in your commit message, I've shared a Udacity git commit message styleguide earlier. Please refer that again.

File: articles/File_download.md (hunk 5 above)
Comment (sharma-shruti) [reply articles/File_download.md:10]:
```powershell
function Save-Video {
        [Cmdletbinding()]
        param(
                [Parameter(Mandatory=$true)]
                [String]$urlString
        )
        $videoId = ([System.Uri]$urlString).Query.Split("=")[1]
        $videoUrl = "https://fast.wistia.net/embed/iframe/" + $videoId
        youtube-dl $videoUrl
        $outputFile = Get-ChildItem *.bin -Name
        $fileString = $outputFile.Split("-")[0]+"-$videoId" + ".mp4"
        $renameFile = mv $outputFile $fileString
        rclone copy *.mp4 remote:/Videos/ -v
}
```

Comment (abhi18av) [reply articles/File_download.md:10]: Cool 👍 

Let's get this rolling forward and then add changes later.

File: articles/File_download.md (hunk 3 above)
Comment (abhi18av) [reply articles/File_download.md]: Well, think of this as a separate box. Perhaps, it's best to discuss this on call :)

**Pull request merged**